plugin/src/main/groovy/com/novoda/staticanalysis/EvaluateViolationsTask.groovy

```diff
@@ -1,14 +1,15 @@
 package com.novoda.staticanalysis
 
 import com.novoda.staticanalysis.internal.Violations
-import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.internal.ConventionTask
 import org.gradle.api.tasks.TaskAction
 
-class EvaluateViolationsTask extends DefaultTask {
+class EvaluateViolationsTask extends ConventionTask {
```

Comment:
I understand your concern about the `internal` package, but I used `ConventionTask` because it's extending `DefaultTask`, not the other way around. Maybe Gradle works its magic and adds a convention map to any task anyway? I can definitely try to keep `DefaultTask` here